kol/simplify_all.py

```diff
@@ -0,0 +1,86 @@
+"""Defines functions for simplifying all meshes in a urdf file."""
+
+import logging
+import xml.etree.ElementTree as ET
+from pathlib import Path
+
+import open3d as o3d
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def parse_urdf(file_path: Path) -> ET.Element:
+    with open(file_path, "r") as file:
+        urdf_xml = file.read()
+    if urdf_xml.startswith("<?xml"):
+        urdf_xml = urdf_xml.split("?>", 1)[1].strip()
+    return ET.fromstring(urdf_xml)
+
+
+def simplify_mesh(filepath: str, voxel_size: float) -> str:
+    """Simplifies a mesh by clustering vertices."""
+    mesh = o3d.io.read_triangle_mesh(filepath)
+    simple_mesh = mesh.simplify_vertex_clustering(
+        voxel_size=voxel_size,
+        contraction=o3d.geometry.SimplificationContraction.Average,
+    )
+    logger.info("Simplified mesh from %d to %d vertices", len(mesh.vertices), len(simple_mesh.vertices))
+    # Remove old mesh and save the simplified one
+    ext = Path(filepath).suffix
+    if ext.lower() == ".stl":
+        simple_mesh.compute_vertex_normals()
+    new_filepath = filepath.replace(f"{ext}", f"_simple{ext}")
```

Comment:
upon reflection this will mess up for something like "ben.stl.stl" so we should change to

```
filepath = Path(filepath)
new_filepath = filepath.parent / f"{filepath.stem}_simple{filepath.suffix}"
```

can refactor other parts of the code accordingly